src/odemis/driver/technolution.py

```diff
@@ -250,55 +320,62 @@ def _stopCalibration(self):
         scanner.scanDelay.unsubscribe(self._startCalibration)
         scanner.scanOffset.unsubscribe(self._startCalibration)
 
+        self._calibrationParameters = None
         self.ASM_API_Post_Call("/scan/stop_calibration_loop", 204)
 
     def _setCalibrationMode(self, mode):
+        """
+        Setter for the calibration mode, this method calls the _startCalibration and _stopCalibration methods
+        (responsible for starting/stopping the calibration nd subscriptions to VA's to ensure updated parameters),
+        changes the default scanner and descanner gain values for calibration, and unsubscribes all the listeners
+        from the mppc.dataflow.
+
+        :param mode (bool):
+        """
         if mode:
-            # TODO K.K. setup standard VA values for calibration mode
+            if self._mppc.data._count_listeners() != 0:
+                logging.warning("Unsubscribing all the listeners from the mppc.dataflow, calibration mode cannot be "
+                                "started if the dataflow still has subscribers.")
+                while self._mppc.data._count_listeners() != 0:
+                    try:
+                        self._mppc.data._listeners.pop()
+                    except:
+                        # If removing one of the listeners fails that is probably due to a lack of listeners
+                        break
```

Comment:
This is not something you should do. That's why the function and attributes are *private*. Actually, DataFlow also has a `_remote_listeners`, so that function will not work.
Instead, I'd suggest to either have just the warning and fail, or just explicitly stop the acquisition.

Reply:
Stopping the acquisition would indeed be nicer. Currently if we cancel the acquisition we do not also remove all the listeners, should we maybe also implement that?
Here I like to drop the listeners because in calibration mode they are subscribed to a totally different stream of data than in acquisition mode.
@Sabrina85 What do you think?

Reply:
Just check the state of the acquisition and assume that the subscribers are handled